modules/internal/ChapelLocale.chpl

```diff
@@ -48,8 +48,29 @@ module ChapelLocale {
     // The parent of the root locale is nil (by definition).
     const parent : locale;
 
-    // To be removed from the required interface once legacy code is adjusted.
-    // Modified in RootLocale.init().
+    pragma "no doc" var nPUsPhysAcc: int;    // HW cores, OS accessible
+    pragma "no doc" var nPUsPhysAll: int;    // HW cores, all
+    pragma "no doc" var nPUsLogAcc: int;     // HW threads, OS accessible
+    pragma "no doc" var nPUsLogAll: int;     // HW threads, all
+
+    /*
+      This returns the number of instances of the CPU architecture on
+      this locale.  If ``physical==true`` this is the number of physical
+      PUs ("cores").  Otherwise, it is the number of hardware threads,
+      which on a multithreaded hardware architecture would be larger.
+      If ``accessible==true`` then the count reflects any OS limitations
+      on which PUs may be accessed.  If ``accessible==false`` then all
+      CPUs that seem to exist are included.
+     */
+    inline
+    proc numPUs(physical: bool = true, accessible: bool = true)
```

Comment:
I find this side conversation (which was hidden by default due to the diff being outdated) intriguing, but think we should focus on the first-order concern about whether there should be a default (and if so, what it should be) first.

If my "have the default track whether we're in logical or physical mode" proposal held up, a possible third option for an enum could be "track the default".  But generally, I think that enums are a bit clumsy for cases like this and would probably still be inclined to keep it a bool and just make the default expression more complex rather than introducing a third option.